cmd/addon-operator-manager/main.go

```diff
@@ -253,6 +255,8 @@ func setup() error {
 		if !featuretoggle.IsEnabled(featureToggleHandler, addonOperatorObjectInCluster) {
 			continue
 		}
+		// TODO: what does this do? It looks to me like it updates featureToggleHandler.AddonReconcilerOptsToUpdate
+		// but then AddonReconcilerOptsToUpdate isn't used anywhere
```

Comment:
Actually, this method is used to attach the manager's scheme and client available to the variable `addonReconcilerOptions`.

so before line number 260, the value of `addonReconcilerOptions = []addoncontroller.AddonReconcilerOptions{}`

and after line number 260,
```go
addonReconcilerOptions = []addoncontroller.AddonReconcilerOptions{
    addoncontroller.WithMonitoringStackReconciler{
	    	Client: mgr.GetClient(),
		    Scheme: mgr.GetScheme(),
	    },
    addoncontroller.WithPackageOperatorReconciler{
	    	Client: mgr.GetClient(),
		    Scheme: mgr.GetScheme(),
	    },
}
```

This slice consists of objects where each object implements two methods: `ApplyToAddonReconciler` and `ApplyToControllerBuilder`. ([Check this out](https://github.com/openshift/addon-operator/blob/main/internal/controllers/addon/addon_reconciler_options.go#L11-L14))

Both of these methods are executed just before the startup of addon-operator, so that depending on which feature-toggle is enabled, the [sub-reconcilers](https://github.com/openshift/addon-operator/blob/main/internal/controllers/addon/controller.go#L125-L127) are accordingly setup and [controller-builder](https://github.com/openshift/addon-operator/blob/main/internal/controllers/addon/controller.go#L242-L244) is accordingly setup as well.